frontend/src/Components/ShippingInfo/ShippingInfo.jsx

```diff
@@ -1,85 +1,231 @@
-import React, { useState } from 'react';
-import './shippinginfo.css';
-import { useDispatch, useSelector } from 'react-redux';
-import { addBookSelector } from '../../Redux/selector';
-import { Link } from 'react-router-dom';
-import { setShippingInfo } from '../../Redux/actions';
+import React, { useState, useEffect } from "react";
+import "./shippinginfo.css";
+import { useDispatch, useSelector } from "react-redux";
+import { addBookSelector } from "../../Redux/selector";
+import { Link, useNavigate } from "react-router-dom";
+import { setShippingInfo } from "../../Redux/actions";
 
 function ShippingInfo() {
-    const dispatch = useDispatch();
-    const cartBooks = useSelector(addBookSelector);
+  const navigate = useNavigate();
+  const dispatch = useDispatch();
+  const cartBooks = useSelector(addBookSelector);
 
-    // State để lưu trữ thông tin vận chuyển
-    const [shippingDetails, setShippingDetails] = useState({
-        recipientName: '',
-        companyName: '',
-        streetAddress: '',
-        landmark: '',
-        country: 'VietNam',
-        cityName: '',
-        zipCode: '',
-        mobile: '',
-        phone: ''
+  // State to store shipping details and errors
+  const [shippingDetails, setShippingDetails] = useState({
```

Comment:
Should create custom hook for shipping and declare all function which related to shipping